engine/class_modules/sc_death_knight.cpp

```diff
@@ -2666,6 +2674,122 @@ struct magus_pet_t : public death_knight_pet_t
   }
 };
 
+struct reanimated_shambler_pet_t : public death_knight_pet_t
+{
+  struct necroblast_t : public pet_action_t<reanimated_shambler_pet_t, melee_attack_t>
+  {
+    necroblast_t( reanimated_shambler_pet_t* player ) :
+      pet_action_t( player, "necroblast", player -> find_spell( 334851 ) )
+    {
+    }
+
+    void execute() override
+    {
+      pet_action_t::execute();
+
+      p()->dismiss();
+    }
+
+    void impact( action_state_t* state ) override
+    {
+      pet_action_t::impact( state );
+      p() -> o() -> trigger_festering_wound( state, 1, p() -> o() -> procs.fw_necroblast );
+    }
+
+    double calculate_direct_amount( action_state_t* state ) const override
+    {
+      if ( p() -> o() -> bugs )
+      {
+        // See https://github.com/SimCMinMax/WoW-BugTracker/issues/725
+        // we think the total ap for necroblast is (total_ap + base_amount) * ap_coefficient
+        // action_t::calculate_direct_amount gets the total ap with base_amount + ap_coefficient * total_ap
+        // because of this, I need to directly modify the ap for the action instead of setting the base damage in
+        // the necroblast constructor as the latter method throws off the calcuations and doesn't reflect in game values
+        state->attack_power += 150;
+      }
+
+      double da = pet_action_t::calculate_direct_amount( state );
+
+      return da;
+    }
+  };
```

Comment:
My gut says reaching out and modifying the state like this is bad but I'm not sure how to get around it.  I'll take any suggestions on how to do this better.  Based on the math from the spreadsheet, the extra base damage seems to be added to the AP amount before multiplying by the coefficient.  So I need to modify the AP of just this action.  I validated that only the damage numbers from necroblast are affected by this.  At this point just hoping Blizzard fixes it.

Reply:
Yeah as long as it's reasonably close to the damage seen ingame and it doesn't negatively affect the rest of the simulation it's fine.